targets/frontend/src/components/list/DataList.tsx

```diff
@@ -0,0 +1,146 @@
+import * as React from "react";
+import { useState } from "react";
+import Box from "@mui/material/Box";
+import Table from "@mui/material/Table";
+import TableBody from "@mui/material/TableBody";
+import TableCell from "@mui/material/TableCell";
+import TableContainer from "@mui/material/TableContainer";
+import TableRow from "@mui/material/TableRow";
+import Checkbox from "@mui/material/Checkbox";
+import { Toolbar } from "./Toolbard";
+import { Head } from "./Head";
+import { Data, HeadCell, Source } from "./type";
+import { PublishModal } from "./PublishModal";
+
+type Props<T extends Data> = {
+  source: Source;
+  readonly headCells: HeadCell<T>[];
+  rows: T[];
+  onClickItem: (id: string) => void;
+  onClickCreation: () => void;
+  setSearch: (value: string | undefined) => void;
+  customFilters?: React.ReactNode;
+};
+
+type ItemCheck = {
+  [id: string]: boolean;
+};
+
+export const DataList = <T extends Data>({
+  source,
+  headCells,
+  rows,
+  onClickItem,
+  onClickCreation,
+  setSearch,
+  customFilters,
+}: Props<T>) => {
+  const [itemsCheck, setItemCheck] = useState<ItemCheck>({});
+  const [isOpen, showModal] = useState<boolean>(false);
+
+  const handleSelectAllClick = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setItemCheck(
+      rows.reduce((acc, item) => {
```

Comment:
Tu peux passer les typages en générique dans le reduce, je trouve ca plus sympa qu'un `as`
```suggestion
      rows.reduce<ItemCheck>((acc, item) => {
```